File: frontend/src/types/report.ts
export interface Report {
  id: number;
  title: string;
  description?: string;
  region: string;
  institution?: string;
  budget_reported?: number;
  portions_reported?: number;
  document_url?: string;
  ocr_extracted_text?: string;
  created_at: string;
  updated_at?: string;
}

export interface ReportListResponse {
  items: Report[];
  total: number;
  page: number;
  size: number;
}

File: backend/app/schemas/report.py
"""Report schemas — request/response DTOs for official reports."""

from datetime import datetime
from pydantic import BaseModel


class ReportBase(BaseModel):
    """Shared official report fields."""
    title: str
    description: str | None = None
    region: str
    institution: str | None = None
    budget_reported: float | None = None
    portions_reported: int | None = None


class ReportCreate(ReportBase):
    """Schema for creating an official report."""
    pass


class ReportResponse(ReportBase):
    """Schema for official report API responses."""
    id: int
    document_url: str | None = None
    ocr_extracted_text: str | None = None
    created_at: datetime
    updated_at: datetime | None = None

    class Config:
        from_attributes = True


class ReportListResponse(BaseModel):
    """Paginated list of reports."""
    items: list[ReportResponse] = []
    total: int = 0
    page: int = 1
    size: int = 20
